core/templates/pages/collection-editor-page/editor-tab/collection-node-creator.component.ts

```diff
@@ -79,7 +80,9 @@ export class CollectionNodeCreatorComponent {
 
         if (summaryBackendObject) {
           this.collectionLinearizerService.appendCollectionNode(
-            this.collection, newExplorationId, summaryBackendObject);
+            this.collection, newExplorationId,
+            summaryBackendObject as unknown as
+            LearnerExplorationSummaryBackendDict);
```

Comment:
We need arg of type LearnerExplorationSummaryBackendDict.

export interface LearnerExplorationSummaryBackendDict {
  'category': string;
  'community_owned': boolean;
  'activity_type': string;
  'last_updated_msec': number;
  'ratings': ExplorationRatings;
  'id': string;
  'created_on_msec': number;
  'human_readable_contributors_summary': HumanReadableContributorsSummary;
  'language_code': string;
  'num_views': number;
  'objective': string;
  'status': string;
  'tags': string[];
  'thumbnail_bg_color': string;
  'thumbnail_icon_url': string;
  'title': string;
}

But passing arg is type 
export interface ExplorationSummaryDict {
  'category': string;
  'community_owned': boolean;
  'human_readable_contributors_summary': (
    HumanReadableContributorsSummary);
  'id': string;
  'language_code': string;
  'num_views': number;
  'objective': string;
  'status': string;
  'tags': [];
  'thumbnail_bg_color': string;
  'thumbnail_icon_url': string;
  'title': string;
}

This function actually working as adding collection node, means append new exploration. But types are almost same but missing some properties. Missing property is useful in collection. Need help in it.

Reply:
Does both the interface serves the same purpose? It can be the case that `ExplorationSummaryDict` can be replaced with `LearnerExplorationSummaryBackendDict` because it is a relatively newer interface with the same properties apart from the left four and might be added by mistake at the point of migration. Do check if extra properties present in `LearnerExplorationSummaryBackendDict` are also used at places where `ExplorationSummaryDict` is being used

Reply:
It is not used in, where `ExplorationSummaryDict` is used. Properties like `last_updated_msec` is just used inside `LearnerExplorationSummaryBackendDict`. I think we need some refactoring in frontend to set intial values to them when exploration is used in collections.

Reply:
Okay, can you try refactoring it?

Reply:
Removed this file for now, will take a look on later stage.